report/renderer/i18n.js

```diff
@@ -22,25 +22,58 @@ export class I18n {
     // When testing, use a locale with more exciting numeric formatting.
     if (locale === 'en-XA') locale = 'de';
 
-    this._numberDateLocale = locale;
-    this._numberFormatter = new Intl.NumberFormat(locale);
-    this._percentFormatter = new Intl.NumberFormat(locale, {style: 'percent'});
+    this._locale = locale;
     this._strings = strings;
   }
 
   get strings() {
     return this._strings;
   }
 
+  /**
+   * @param {number} number
+   * @param {number} granularity
+   * @param {Intl.NumberFormatOptions} opts
+   * @return {string}
+   */
+  _formatNumberWithGranularity(number, granularity, opts = {}) {
+    opts = {...opts};
+    const log10 = -Math.log10(granularity);
+    if (!Number.isFinite(log10) || (granularity > 1 && Math.floor(log10) !== log10)) {
+      throw new Error(`granularity of ${granularity} is invalid`);
+    }
+
+    if (granularity < 1) {
+      opts.minimumFractionDigits = opts.maximumFractionDigits = Math.ceil(log10);
+    }
+
+    number = Math.round(number / granularity) * granularity;
+
+    // Avoid displaying a negative value that rounds to zero as "0".
+    if (Object.is(number, -0)) number = 0;
+
+    return new Intl.NumberFormat(this._locale, opts).format(number).replace(' ', NBSP2);
```

Comment:
It seems like it should be fine to do the replace thing, though it feels dirty and maybe there's somehow an issue with rtl or something? 🤷 

Do we still need it? I assume we do, but it would be cool if our various layout initiatives over the years made it unnecessary to keep them from breaking across lines

Reply:
Interestingly, for `new Intl.NumberFormat('fr', {style: 'unit', unit: 'second', unitDisplay: 'long'}).format(5)` (`'5 secondes'`), the second character *is* NBSP2, but that doesn't appear true for most (all?) other languages.

`new Intl.NumberFormat('fr', {style: 'unit', unit: 'second', unitDisplay: 'short'})` has `\u202f` (narrow no-break space) as its second character.

Also, languages disagree on spacing of units :/
`new Intl.NumberFormat('en-US', {style: 'unit', unit: 'byte', unitDisplay: 'narrow'}).format(5)` -> `'5B'`
`new Intl.NumberFormat('en-US', {style: 'unit', unit: 'byte', unitDisplay: 'narrow'}).format(5)` -> '5 B'

Reply:
omg this is infuriatingly inconsistent. no way this is due to locale differences, these control characters are formatting and what locale would expect numbers to be disassociated from their unit by newlines?? this _has_ to be a mistake in the localization library...

your `fr` example is odd because the CLDR data has 32 space for [long](https://github.com/unicode-org/cldr/blob/main/common/main/fr.xml#L10203) but `narrow non-break space` for [short](https://github.com/unicode-org/cldr/blob/main/common/main/fr.xml#L11352), based on `Intl` output I'd expect the former to show NBSP2. (narrow has no whitespace)

One can sort of see reason for there being a non-breaking space between the number and the short string but not for the longer one, as without that number the shorter strings being further from their number gives them far less context.

Reply:
> One can sort of see reason for there being a non-breaking space between the number and the short string but not for the longer one, as without that number the shorter strings being further from their number gives them far less context.

totally, but why isn't it the same across languages?? Or at least, why wouldn't english (or german or spanish or...) do it as well?

> (your last example is the same two lines of code btw. i'd expect narrow to never include any whitespace)

oops, I meant `de` for the second one:

`new Intl.NumberFormat('en-US', {style: 'unit', unit: 'byte', unitDisplay: 'narrow'}).format(5)` -> `'5B'`
`new Intl.NumberFormat('de', {style: 'unit', unit: 'byte', unitDisplay: 'narrow'}).format(5)` -> `'5 B'`

Reply:
> Do we still need it? I assume we do, but it would be cool if our various layout initiatives over the years made it unnecessary to keep them from breaking across lines

<img width="750" alt="image" src="https://user-images.githubusercontent.com/4071474/163394142-cc5b6a75-595c-4768-a078-074ebfdf83cf.png">

hmm....